### ISSUE TITLE
import csep failing 

### ISSUE THREAD

**Issue by wsavran** [SCECcode/pycsep#168]
using numpy=1.22.0 and obspy 1.2.2 we get the following error when trying to import pycsep 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/csep/__init__.py", line 7, in <module>
    from csep.core import forecasts
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/csep/core/forecasts.py", line 14, in <module>
    from csep.core.catalogs import AbstractBaseCatalog
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/csep/core/catalogs.py", line 18, in <module>
    from csep.utils.comcat import SummaryEvent
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/csep/utils/comcat.py", line 19, in <module>
    from obspy.core.event import read_events
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/__init__.py", line 39, in <module>
    from obspy.core.utcdatetime import UTCDateTime  # NOQA
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/core/__init__.py", line 124, in <module>
    from obspy.core.utcdatetime import UTCDateTime  # NOQA
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/core/utcdatetime.py", line 27, in <module>
    from obspy.core.util.deprecation_helpers import ObsPyDeprecationWarning
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/core/util/__init__.py", line 27, in <module>
    from obspy.core.util.base import (ALL_MODULES, DEFAULT_MODULES,
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/core/util/base.py", line 36, in <module>
    from obspy.core.util.misc import to_int_or_zero, buffered_load_entry_point
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/obspy/core/util/misc.py", line 214, in <module>
    loadtxt(np.array([0]), ndmin=1)
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/numpy/lib/npyio.py", line 1086, in loadtxt
    ncols = len(usecols or split_line(first_line))
  File "/Users/wsavran/Applications/anaconda3/envs/gji-hybrids/lib/python3.10/site-packages/numpy/lib/npyio.py", line 977, in split_line
    line = line.split(comment, 1)[0]
AttributeError: 'numpy.int64' object has no attribute ‘split’
```

After some digging, this looks like an issue from an update to numpy. 

Solution: we should pin versions of main packages for releases to prevent releases from breaking over time. We can push out minor patches of pycsep to bump version numbers of dependencies if needed.

**Comment by wsavran** [comment]
linking relevant issue: https://github.com/obspy/obspy/issues/2912

**Comment by wsavran** [comment]
fixed with #115

**Comment by wsavran** [comment]
re-opening this issue. will be fixed with the v0.5.2 release.

**Comment by mherrmann3** [comment]
`numpy` should be unpinned again from the requirements, because [`obspy v1.3.0`](https://github.com/obspy/obspy/releases/tag/1.3.0) _"removed obsolete wrapper around numpy.loadtxt causing import error with numpy 1.22"_.
In this way we allow pycsep users using Python 3.10 (like me) to update to newer `numpy` (v1.21.6+; currently at v1.23) and `scipy `(v1.8.1+; currently at v1.9.0) versions.
